Chapter_4_Advanced_Structuring_and_Execution.md

```diff
@@ -0,0 +1,599 @@
+- Executing preparation steps to navigate to the feature under test.
+
+Example of defining a default Test|Task Setup in the suite settings and overriding it on a test case:
+
+```robotframework
+*** Settings ***
+Test Setup    Login As Standard User
+
+
+*** Test Cases ***
+User Action Test With Default Setup    # Default Test Setup is applied
+    Perform User Actions    0815
+
+Another User Action With Default Setup    # Default Test Setup is applied
+    Perform another User Action    4711
+
+Admin Access Test With Local Setup
+    [Setup]    Login As Admin    # Override the default setup
+    Perform Admin Actions   007
+
+No Setup Test
+    [Setup]    NONE    # Override and disable the setup by case-sensitive NONE
+    Perform Actions Without Login   000
+```
+
+
+
+### 4.1.3 Keyword Setup
+
+> [!IMPORTANT]
+> LO-XX Recall key characteristics and syntax of Keyword Setup (K1)
+
+A **Keyword Setup** is executed before the body of a user keyword is executed.
+It allows for preparation steps specific to that keyword or ensures that the keyword's requirements are met before execution.
+
+**Key characteristics of Keyword Setup:**
+- Executed before the keyword's body.
+- If the Keyword Setup fails, the keyword's body is not executed.
+- Logged in the execution log as a separate section, indicating the setup status.
+
+**Typical use cases:**
+- Opening connections or files needed by the keyword.
+- Initializing variables or data structures.
+- Ensuring preconditions specific to the keyword are met.
+
+Example of defining a Keyword Setup:
+
+```robotframework
+*** Keywords ***
+Process Data
+    [Setup]    Open Data Connection
+    Process the Data
+```
+
+
+
+
+## 4.2 Teardowns (Suite, Test|Task, Keyword)
+
+> [!IMPORTANT]
+> LO-XX Understand the different levels where and how a Teardowns can be defined and when they are executed (K2)
+>
+> LO-XX Recall the typical use cases for using a Teardowns (K1)
+
+In automation, tests|tasks are typically executed in a linear sequence.
+This linear execution can lead to issues when a preceding test|task fails, potentially affecting subsequent tests|tasks due to an unclean state of the system under test or the automated environment.
+To prevent such issues, Robot Framework provides the **Teardown** functionality, which can be defined at the suite, test|task, or keyword level.
+
+As mentioned before, a failure resulting in a keyword with the status `FAIL` will cause Robot Framework not to execute all subsequent keywords of the current test|task.
+These not-executed keywords will receive the status `NOT RUN`.
+
+A **Teardown** is a single keyword call with potential argument values that is executed after the child suites, test|tasks, and keywords have completed execution, regardless of the outcome, even if previously executed elements have failed.
+It ensures that necessary cleanup actions are performed, maintaining the integrity of the test environment for subsequent executions.
```

Comment:
leave out test, because RPA would be done in a production environment
```suggestion
It ensures that necessary cleanup actions are performed, maintaining the integrity of the environment for subsequent executions.
```